### PR TITLE
Platform Docs: Add trusted by section

### PR DESCRIPTION
**Pull request by youknowriad**
Related #53874 

This PR adds a "trusted by" section to the Gutenberg as a framework docs site. I'm also removing some empty documentation pages for now. We may want to add these back later.

Screenshot 2023-12-04 at 11 33 21 AM

I think this site needs some design love but I'm hoping @jameskoster @jasmussen or @SaxonF could help with some of that in follow-ups.

Check the site by doing

```
cd platform-docs
npm install
npm start
```

### CODE REVIEW

File: platform-docs/src/components/HomepageTrustedBy/index.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	},
+	{
+		title: 'WordPress',
+		img: require( '@site/static/img/wordpress.png' ).default,
+	},
+	{
+		title: 'Day One',
+		img: require( '@site/static/img/dayone.png' ).default,
+	},
+];
```
Comment (youknowriad) [inline platform-docs/src/components/HomepageTrustedBy/index.js:27]: Happy to add more "Gutenberg" users, these are the ones I'm aware of.

**Comment by jameskoster** [comment]
It's always a bit tricky balancing logos like this, but my drive-by feedback would be to try and improve that a bit. Something like:

logos

Using monochrome or grayscale variants could help too, depending on how much attention you want to bring to this section.

**Comment by youknowriad** [comment]
@jameskoster I guess I could use some help with that :P. What tools do you use these days to edit things like that?

**Comment by jameskoster** [comment]
It's probably fine to do with css by adjusting the height for each logo accordingly? If you wanted to copy the mockup above:

* WP = 56px
* Tumblr = 34px
* Day One = 40px

Otherwise here are the resized images, though ideally they should be transparent svgs:

dayone
tumblr
wp

**Comment by youknowriad** [comment]
@jameskoster Are you sure about these heights? Did you use the same images, cause I got something completely different :P ? I can try to approximate that though.

**Comment by youknowriad** [comment]
I pushed a commit, here is how it looks now

Screenshot 2023-12-04 at 1 50 49 PM

**Comment by jameskoster** [comment]
> Did you use the same images

Apologies, no, I cropped them.

> I pushed a commit, here is how it looks now

That looks okay to me, outside of diving into the code and replacing the images with svgs which regrettably I don't have time to do right now.

File: platform-docs/src/components/HomepageTrustedBy/index.js (hunk 1 above)
Comment (tellthemachines) [reply platform-docs/src/components/HomepageTrustedBy/index.js:27]: The [Drupal extension](https://www.drupal.org/project/gutenberg) might be worth a mention?

Comment (youknowriad) [reply platform-docs/src/components/HomepageTrustedBy/index.js:27]: I thought about it but at the moment the Drupal doesn't really use the block editor package as a framework but they reuse the whole plugin somehow. I know they're working on it to improve the integration so I left it out for now.

**Review by ntsekouras — APPROVED** [review]
Let's get this in and iterate with polishing if needed.

**Pull request merged**